features/update/deduplication/feature.py

```diff
@@ -0,0 +1,62 @@
+import asyncio
+from datetime import timedelta
+
+from temporalio import activity, workflow
+from temporalio.client import WorkflowHandle
+
+from harness.python.feature import Runner, register_feature
+
+
+@workflow.defn
+class Workflow:
+    counter = 0
+
+    @workflow.run
+    async def run(self) -> str:
+        await workflow.wait_condition(lambda: self.counter == 2)
+        return "Hello, World!"
+
+    @workflow.update
+    async def inc_counter(self):
+        self.counter += 1
+        # Verify that dedupe works pre-update-completion
+        await asyncio.sleep(1)
```

Comment:
Mentioned elsewhere too so I'll stop, but in general would like this to wait on caller stimulus instead of a timer for flake/perf reasons